pkg/rdt/prometheus.go

```diff
@@ -100,19 +101,30 @@ func (c *collector) describeL3(feature string) *prometheus.Desc {
 	return d
 }
 
-func (c *collector) collectGroupMetrics(ch chan<- prometheus.Metric, mg MonGroup) {
-	allData := mg.GetMonData()
-
+func (c *collector) collectMonGroupMetrics(ch chan<- prometheus.Metric, mg MonGroup) {
 	annotations := mg.GetAnnotations()
 	customLabelValues := make([]string, len(customLabels))
 	for i, name := range customLabels {
 		customLabelValues[i] = annotations[name]
 	}
 
+	c.collectGroupMetrics(ch, mg, customLabelValues...)
+}
+
+func (c *collector) collectGroupMetrics(ch chan<- prometheus.Metric, g ResctrlGroup, customLabels ...string) {
+	allData := g.GetMonData()
+
+	cgName, mgName := "", ""
+	if mg, ok := g.(MonGroup); ok {
+		cgName = mg.Parent().Name()
+		mgName = mg.Name()
+	} else {
+		cgName = g.Name()
+	}
+
 	for cacheID, data := range allData.L3 {
+		labels := append([]string{cgName, mgName, fmt.Sprint(cacheID)}, customLabels...)
```

Comment:
nit: Is that fmt.Sprintf(cacheID) really necessary here ? With a goroutine I would understand, but here ?

Reply:
We need to convert the uint64 cacheID to string. Would you prefer strconv.FormatUint() or did you have something else in mind?